src/antsibull_docs/lint_plugin_docs.py

```diff
@@ -311,6 +311,82 @@ def _create_lookup(
     return lookup
 
 
+class _ValidationWalker(dom.NoopWalker):
+    def __init__(self, markup_validator: "_MarkupValidator", key: str):
+        self.markup_validator = markup_validator
+        self.errors = markup_validator.errors
+        self.key = key
+
```

Comment:
Maybe a helper method could be useful here to avoid some duplication? Something like:

```
def _append_error(self, source: str, message: str) -> None:
        self.errors.append(f"{self.key}: {source}: {message}")
```

Reply:
Changed in 340cdb21a86d777563f2705163cd7838e12ba558.

Reply:
:raised_hands: I think I'm going to print this and frame it. I finally made a useful code-related suggestion to one of your PRs.